SiLi rewrite relating to issue #559

jsmallcombe opened the pull request:
Attempted rewrite of Sili and s3 to clean things up and bring in line.

r3dunlop commented on include/TS3Data.h:
```diff
@@ -21,11 +23,16 @@ class TS3Data : public TGRSIDetectorData  {
     std::vector<UShort_t>  fS3_SectorNumber;
     std::vector<TFragment> fS3_SectorFragment;
+    
+   static bool fIsSet;      //!
```
We have found that bool's in root take an entire byte. I will leave this for now, but you should make a Short so you have effectively 8 bools you can save for the price of one (checkout griffin crystal and pileup to see what I mean).

jsmallcombe replied:
I'm happy to follow the efficient style ... but this is the same in griffindata as s3data
"bool TGriffinData::fIsSet = false;"

Could you highlight where you mean static variables? Its possible I copied something wrong when trying to match other detectors.

r3dunlop replied:
My bad. You are right. This is in the data so all of this is irrelevant. For some reason I thought this was the hit. Sorry.

> On Oct 29, 2015, at 4:43 PM, jsmallcombe <notifications@github.com> wrote:
> 
> In include/TS3Data.h <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#discussion_r43453909>:
> 
> > @@ -21,11 +23,16 @@ class TS3Data : public TGRSIDetectorData  {
> >      std::vector<UShort_t>  fS3_SectorDetector;
> >      std::vector<UShort_t>  fS3_SectorNumber;
> >      std::vector<TFragment> fS3_SectorFragment;
> > +    
> > +   static bool fIsSet;      //!
> I'm happy to follow the efficient style ... but this is the same in griffindata as s3data
> "bool TGriffinData::fIsSet = false;"
> 
> Could you highlight where you mean static variables? Its possible I copied something wrong when trying to match other detectors.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/GRIFFINCollaboration/GRSISort/pull/560/files#r43453909>.
>

r3dunlop commented:
This just needs the SetDetector and SetPosition calls to be removed during building. (Otherwise my fix won't work). I'll merge this, it's just a heads up.

r3dunlop commented:
Also, for the position stuff, we do it through calls to a private "GetChannelPosition" in the derived detector hit classes.

jsmallcombe commented:
Ok I think that's all working now.

jsmallcombe commented:
For our pulse fitting class I implemented some draw functions and made some small improvement tweaks as well as fixing a stupid error that was making results meaningless.
We have also added the SFU fit to TRF which seems to be producing better results.

pcbend commented:
Can you please explain how trf is different and quantify better?
On Oct 30, 2015 5:15 PM, "jsmallcombe" <notifications@github.com> wrote:

> For our pulse fitting class I implemented some draw functions and made
> some small improvement tweaks as well as fixing a stupid error that was
> making results meaningless.
> We have also added the SFU fit to TRF which seems to be producing better
> results.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-152677449>
> .
>

jsmallcombe commented:
We have added a TPulseAnalyzer fit to TRF to create an alternative value for the phase.

That alternative value appears to work in our analysis, where the other didnt.

pcbend commented:
Again , can you please descibe how this is different and quantify the
results?

The rf fitter as in the code is already Kris Starostas fitting routine and
matched work done by the sfu group over the past years.  It is hard to
imagine this routine changing as it is just fitting a sine wave.
Additionally as there is current analysis that depends on these results
please quantify how the results are better.  Time spectra can be made
easily between the current value and the timing seen in the other tig-10
digitizers.  How do these spectra change ? What was the need to change the
result?  Does some quanity like the peak to background actually change?  If
so, how much?
On Oct 30, 2015 5:31 PM, "jsmallcombe" <notifications@github.com> wrote:

> We have added a TPulseAnalyzer fit to TRF to create an alternative value
> for the phase.
>
> That alternative value appears to work in our analysis, where the other
> didnt.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-152679832>
> .
>

jsmallcombe commented:
@jhenderson88

jhenderson88 commented:
Not necessarily better, but more consistent with the other SFU-style fits in terms of the results.

pcbend commented:
That I can see and understand the merit.  The word better early really made
me wonder.  Do they give the same reault than? Has this been tested?
On Oct 30, 2015 5:53 PM, "jhenderson88" <notifications@github.com> wrote:

> Not necessarily better, but more consistent with the other SFU-style fits
> in terms of the results.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-152681477>
> .
>

jhenderson88 commented:
No, it hasn't been tested yet. Currently we are able to reproduce the SFU plots fairly well with their style of RF fit (unsurprisingly). This gives me some confidence that our other fits are doing their job. Once I have everything implemented properly I will perform a quantitative test.

pcbend commented:
Awesome.  Having things done uniformly will be very nice.
On Oct 30, 2015 5:59 PM, "jhenderson88" <notifications@github.com> wrote:

> No, it hasn't been tested yet. Currently we are able to reproduce the SFU
> plots fairly well with their style of RF fit (unsurprisingly). This gives
> me some confidence that our other fits are doing their job. Once I have
> everything implemented properly I will perform a quantitative test.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-152681877>
> .
>

pcbend commented:
I would really like to see the results of the rf before this becomes
permanent
On Nov 2, 2015 10:42 AM, "Ryan Dunlop" <notifications@github.com> wrote:

> Merged #560 <https://github.com/GRIFFINCollaboration/GRSISort/pull/560>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#event-452102889>
> .
>

r3dunlop commented:
Do you want it reverted then until it's tested?

pcbend commented:
Can we just remove the rf part till tested?  The issue is its currently
being used to make timing gates in the Sr and I believe csm analysis.  I
hate for things to turn up empty because we changed the way some thing
works.
On Nov 2, 2015 11:12 AM, "Ryan Dunlop" <notifications@github.com> wrote:

> Do you want it reverted then until it's tested?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-153068180>
> .
>

pcbend commented:
Though I should say I have no proof either way, just like to see it behave
the same.
On Nov 2, 2015 11:20 AM, "Peter" <pcbend@gmail.com> wrote:

> Can we just remove the rf part till tested?  The issue is its currently
> being used to make timing gates in the Sr and I believe csm analysis.  I
> hate for things to turn up empty because we changed the way some thing
> works.
> On Nov 2, 2015 11:12 AM, "Ryan Dunlop" <notifications@github.com> wrote:
>
>> Do you want it reverted then until it's tested?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-153068180>
>> .
>>
>

r3dunlop commented:
No I don't think there is a way to selectively add commits.

r3dunlop commented:
Hmm turns out a revert means you have to do another pull request as well. That's unfortunate. @jsmallcombe can you do another request, and show that the Rf works? We'll merge it then. I'll try to stay out of the Tigress stuff unless it's about a bug that I can try to fix.

jsmallcombe commented:
I dont know what you want me to "show" you. The RF does work, @jhenderson88 is using it.
The issue of why/if the old RFFitter is not working correctly for us is a different issue.

jhenderson88 commented:
The original RF hasn't been removed, an alternative fitting routine has just been added, so this shouldn't affect anything using the RFFitter routine.

pcbend commented:
A start would be the things discussed earlier in this thread.  The idea is
to show that things work the same or better as previously claimed.

The old rf was working, as also said earlier by jack in this same thread.
It's fine to change things for style arguments but we should  also compare
to previous results.
On Nov 2, 2015 11:48 AM, "jsmallcombe" <notifications@github.com> wrote:

> I dont know what you want me to "show" you. The RF does work,
> @jhenderson88 <https://github.com/jhenderson88> is using it.
> The issue of why/if the old RFFitter is not working correctly for us is a
> different issue.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-153079534>
> .
>

pcbend commented:
I would prefer there to just be one phase in the rf, that can come from
either method as the results should be the same.
On Nov 2, 2015 11:55 AM, "Peter" <pcbend@gmail.com> wrote:

> A start would be the things discussed earlier in this thread.  The idea is
> to show that things work the same or better as previously claimed.
>
> The old rf was working, as also said earlier by jack in this same thread.
> It's fine to change things for style arguments but we should  also compare
> to previous results.
> On Nov 2, 2015 11:48 AM, "jsmallcombe" <notifications@github.com> wrote:
>
>> I dont know what you want me to "show" you. The RF does work,
>> @jhenderson88 <https://github.com/jhenderson88> is using it.
>> The issue of why/if the old RFFitter is not working correctly for us is a
>> different issue.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-153079534>
>> .
>>
>

jsmallcombe commented:
@pcbend the things you asked for earlier in the thread were for differences. @r3dunlop asked me to show that the RF works. These are different requests.

I have created another pull with code commented out, I believe this code should have no affect on people already using the previous method while @jhenderson88 works on showing the differences.

pcbend commented:
I think the re-working and the differences between a working rf are really
the same thing.  I stand by that it is confusing to have to phases.  A
phase, derived from the sfu waveform routine written by me and an phasesfu
also derived from the sfu waveform fitter seem like the definition of
duplicated code.  If they are the same, we should remove one.  If they are
not the same, we should understand why.  As stated earlier, it is hard to
understand how a sine wave fitter could change, I am willing to bet the
results of phase and phasesfu are similar possible different by pi
depending on the choice of the amplitude sign.

On Mon, Nov 2, 2015 at 12:03 PM, jsmallcombe <notifications@github.com>
wrote:

> @pcbend <https://github.com/pcbend> the things you asked for earlier in
> the thread were for differences. @r3dunlop <https://github.com/r3dunlop>
> asked me to show that the RF works. These are different requests.
>
> I have created another pull with code commented out, I believe this code
> should have no affect on people already using the previous method while
> @jhenderson88 <https://github.com/jhenderson88> works on showing the
> differences.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-153083996>
> .
>

jsmallcombe commented:
The question if thing A works and if thing A=B are two different questions.
That is not up for debate.

Sure, duplicate code is bad. @jhenderson88 will show if it is necessary. But I am only concerned with if the new code I am pushing works.

r3dunlop commented:
I think we should just decide on the best one, whatever best means. If we have two, and there is a slight difference, someone down the line may not pay attention to that and get something wrong. I think that’s the only issue here. I don’t care what you guys decide, I just don’t want to be the one responsible for a mistake. Also, since I know nothing about Tigress, I’ll leave this one up to you guys.

> On Nov 2, 2015, at 12:14 PM, jsmallcombe <notifications@github.com> wrote:
> 
> The question if thing A works and if thing A=B are two different questions.
> That is not up for debate.
> 
> Sure, duplicate code is bad. @jhenderson88 <https://github.com/jhenderson88> will show if it is necessary. But I am only concerned with if the new code I am pushing works.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-153087757>.
>

pcbend commented:
I agree with Ryan

The issue here is there is only one correct answer for the phase.... two if
one allows for negative amplitudes, but that will only be off by a constant
pi. If the two functions do not produce the same answer, (i.e. A != B), we
should use the correct one. And I agree this fact is not up for debate.  If
they both produce the same answer, we should use the one that looks and
feels like the other fitting routines (i.e. TPulseAnalyzer)

On Mon, Nov 2, 2015 at 12:17 PM, Ryan Dunlop <notifications@github.com>
wrote:

> I think we should just decide on the best one, whatever best means. If we
> have two, and there is a slight difference, someone down the line may not
> pay attention to that and get something wrong. I think that’s the only
> issue here. I don’t care what you guys decide, I just don’t want to be the
> one responsible for a mistake. Also, since I know nothing about Tigress,
> I’ll leave this one up to you guys.
>
> > On Nov 2, 2015, at 12:14 PM, jsmallcombe <notifications@github.com>
> wrote:
> >
> > The question if thing A works and if thing A=B are two different
> questions.
> > That is not up for debate.
> >
> > Sure, duplicate code is bad. @jhenderson88 <
> https://github.com/jhenderson88> will show if it is necessary. But I am
> only concerned with if the new code I am pushing works.
> >
> > —
> > Reply to this email directly or view it on GitHub <
> https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-153087757
> >.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/560#issuecomment-153088587>
> .
>
